sentence_transformers/sparse_encoder/models/IDF.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import TYPE_CHECKING, Any
+
+import torch
+from safetensors.torch import load_file as load_safetensors_file
+from safetensors.torch import save_model as save_safetensors_model
+from transformers import AutoTokenizer
+
+if TYPE_CHECKING:
+    from transformers import PreTrainedTokenizer
+
+
+class IDF(torch.nn.Module):
+    def __init__(
+        self,
+        weight: torch.Tensor,
+        frozen: bool = False,
+        tokenizer: PreTrainedTokenizer | None = None,
+    ):
+        super().__init__()
+        self.weight = torch.nn.Parameter(weight, requires_grad=not frozen)
+        self.frozen = frozen
+        self.word_embedding_dimension = self.weight.size(0)
+        self.tokenizer = tokenizer
+        self.config_keys = ["frozen", "max_seq_length"]
```

Comment:
```suggestion
        self.config_keys = ["frozen"]
```
The config_keys will be passed as to the `__init__`, so this would crash if it's still there